v1/plugin/collector_proxy_test.go

```diff
@@ -46,7 +47,7 @@ func TestGetMetricTypes(t *testing.T) {
 			for _, m := range r.GetMetrics() {
 				tm := fromProtoMetric(m)
 				idx := fmt.Sprintf("%s.%d", tm.Namespace, tm.Version)
-				So(tm.Namespace.String(), ShouldEqual, metricMap[idx].Namespace.String())
+				So(toString(tm.Namespace.Strings()), ShouldEqual, toString(metricMap[idx].Namespace.Strings()))
```

Comment:
I cannot compare that as it'll never pass. The config is gone after conversion. That's expected from the prior discussion with Cody.

```
* /Users/egu/.gvm/pkgsets/go1.7/snap-1.7/src/github.com/intelsdi-x/snap-plugin-lib-go/v1/plugin/collector_proxy_test.go
  Line 51:
  Expected: '"plugin.Metric{Namespace:plugin.namespace{plugin.namespaceElement{Value:\"a\", Description:\"\", Name:\"\"}, plugin.namespaceElement{Value:\"b\", Description:\"\", Name:\"\"}, plugin.namespaceElement{Value:\"c\", Description:\"\", Name:\"\"}}, Version:2, Config:plugin.Config{\"key\":123}, Data:2, Tags:map[string]string{\"tag2\":\"value2\"}, Timestamp:time.Time{sec:63612521385, nsec:588941258, loc:(*time.Location){name:\"\", zone:[]time.zone(nil), tx:[]time.zoneTrans(nil), cacheStart:0, cacheEnd:0, cacheZone:(*time.zone)(nil)}}, Unit:\"int\", Description:\"\", lastAdvertisedTime:time.Time{sec:0, nsec:0, loc:(*time.Location)(nil)}}"'
  Actual:   '"plugin.Metric{Namespace:plugin.namespace{plugin.namespaceElement{Value:\"a\", Description:\"\", Name:\"\"}, plugin.namespaceElement{Value:\"b\", Description:\"\", Name:\"\"}, plugin.namespaceElement{Value:\"c\", Description:\"\", Name:\"\"}}, Version:2, Config:plugin.Config{}, Data:2, Tags:map[string]string{\"tag2\":\"value2\"}, Timestamp:time.Time{sec:63612521385, nsec:588941258, loc:(*time.Location){name:\"\", zone:[]time.zone(nil), tx:[]time.zoneTrans(nil), cacheStart:0, cacheEnd:0, cacheZone:(*time.zone)(nil)}}, Unit:\"int\", Description:\"\", lastAdvertisedTime:time.Time{sec:63612521385, nsec:589213540, loc:(*time.Location){name:\"\", zone:[]time.zone(nil), tx:[]time.zoneTrans(nil), cacheStart:0, cacheEnd:0, cacheZone:(*time.zone)(nil)}}}"'
  (Should resemble)!
```